app/controllers/confirmation_controller.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class ConfirmationController < ApplicationController
+  rescue_from ActiveRecord::RecordNotUnique, with: :error_not_unique
+
+  # This populates instance variables used by the new confirmation form.
+  def new
+    @term = Term.find(params[:term_id])
+    @categories = Category.all
+    @confirmation = Confirmation.new
+  end
+
+  # This receives the submission from the new confirmation form, creating the needed record with the help of various
+  # private methods.
+  def create
+    confirmation = Confirmation.new({
+                                      term_id: params[:term_id],
+                                      user: current_user
+                                    })
+    set_category(confirmation, params[:confirmation][:category])
+    feedback_for(confirmation.save)
+    redirect_to terms_unconfirmed_path
+  end
+
+  private
+
+  # This method takes the result of the confirmation.save directive above and sets an appropriate flash message.
+  #
+  # The final else clause is likely to be difficult to provoke, but I have trouble building an if conditional without
+  # either a default value or an else clause.
+  def feedback_for(result)
+    if result == true && params[:confirmation][:category] == 'flag'
+      flash[:success] = 'Term flagged for review'
+    elsif result == true
+      flash[:success] = "Term confirmed as #{Category.find_by(id: params[:confirmation][:category]).name}"
+    else
+      flash[:error] = 'Unable to finish confirming this term. Please try again, or try a different term.'
+    end
+  end
+
+  # This method catches the RecordNotUnique error in case a duplicate confirmation is ever submitted, and shows an
+  # appropriate error message to the user.
+  def error_not_unique(_exception)
+    flash[:error] = 'Duplicate confirmations are not supported'
+    redirect_to terms_unconfirmed_path
+  end
+
+  # The confirmation form lists options for each Category record, and then one extra option to flag the term for
+  # removal. "Flag" is not a category, but a separate boolean field on the Confirmation model.
```

Comment:
Did you consider having `Flagged for removal` be a category or a boolean on a `Term` itself? (I know that landed in a previous commit and I'm not asking for a change, I'm just curious as something being flagged will be best to be removed from all future queries immediately (to be reviewed by us probably) so putting the flag on Term would make it easy to suppress it from all user facing views via a scope. It could be done also with the flag here, but that requires additional more complex queries any time a Term is being looked up. Not a big deal either way, I'm mostly just curious what was considered when making this decision.

I'm nervous to actually `delete` flagged queries every because by keeping them but suppressing them allows us to always hide them from users whereas deleting them would just allow them to resurface again if someone were ever to search for that term again. I guess our review of suppressed queries might have some thought into how likely it would be to recur and need to be re-suppressed again. Hmmm.

Reply:
I remember discussing a few options, but I'm not sure I could find where now. I do remember thinking about whether we needed 3, 4, or 5 categories - and settling on 4 with the addition of `Undefined`, which humans can use but the knowledge graph does not. A `Flagged` category didn't seem to make sense, based on the assumption that we would actually be deleting terms. If we don't actually delete terms, then perhaps that might make sense to revisit.

My thought about suppressing versus deleting terms is that we shouldn't store records we shouldn't have - keeping them in the database invites problem if the application leaks data or is breached. If a term gets re-submitted, we'll need to delete it again, but I'm not sure it's worth keeping things around in case a duplicate ever gets submitted (remembering how rare repeated searches are in the first place).

I do remember thinking about a flag on the Term, and decided against that approach because having the flag on the Confirmation preserves information about who flagged it - in case follow-up is needed about why they flagged it.

Reply:
Yeah this is complicated. There aren't many reasons we'd want to flag/delete a term and if we hit one I'm nervous to not have a way to prevent it from being shown to all future validators, keep it out of any data exports, etc. I understand your concern about wanting to delete data we shouldn't have, but I'm waffly on whether it's better to know we shouldn't have it (by flagging it as suppressed at the Term level) to make sure it's never included in our work versus having it continue to show up again in the future. I can probably be brought around to your view based on the infrequency of repeat terms, but by deleting things -- contrary to how it may feel -- we'll never actually know for certain we are preventing it from being seen from anyone (other than us as system maintainers who have raw db access). Hmm.